src/test/java/athena/StorageTest.java

```diff
@@ -0,0 +1,72 @@
+package athena;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class StorageTest {
+
+    @Test
+    void saveTaskListData_noPreviousSave_createSaveFile() {
+        Ui ui = new Ui();
+        TaskList taskList = TestSetup.getTestTaskList();
+        Storage storage = new Storage("src/test/java/athena/loadTask.csv", ui);
+        storage.saveTaskListData(taskList);
+        assertTrue(areFilesSame("src/test/java/athena/loadTask.csv", "src/test/java/athena/TaskAnswer.csv"));
+
+
+    }
+
+    private boolean areFilesSame(String file1, String file2) {
+        BufferedReader answer = null;
+        try {
+            answer = new BufferedReader(new FileReader(file1));
+
+
+            BufferedReader attempt = new BufferedReader(new FileReader(file2));
+            String answerLine = answer.readLine();
+            String attemptLine = attempt.readLine();
+            while ((answerLine != null) || (attemptLine != null)) {
+                if (!answerLine.contentEquals(attemptLine)) {
+                    return false;
+                } else {
+                    answerLine = answer.readLine();
+                    attemptLine = attempt.readLine();
+                }
+            }
+            return true;
+        } catch (IOException e) {
+            return false;
+        }
+
+
+    }
+
+    @Test
+    void loadTaskListData_saveFileFound_createTaskList() {
+        Ui ui = new Ui();
+        Storage storage = new Storage("src/test/java/athena/TaskAnswer.csv", ui);
+        TaskList taskList;
+        taskList = storage.loadTaskListData();
+        TaskList tester = TestSetup.getTestTaskList();
+        assertTrue(areTaskListsSame(tester, taskList));
+    }
+
+    private boolean areTaskListsSame(TaskList tester, TaskList other) {
+        if (tester.getMaxIndex() != (other.getMaxIndex())) {
+            return false;
+        } else if (tester.getTasks().equals(other.getTasks())) {
+            return false;
+        } else {
+            return true;
+        }
+
+
+    }
```

Comment:
It would be better override `equals` in `TaskList` instead

https://github.com/AY2021S1-CS2113T-W12-2/tp/blob/30bf904954319f483b762ee3c7f5a378e494c4b7/src/main/java/athena/TaskList.java#L179-L189